### PR TITLE
Update internal actions in manager to take in binary

### PR DESCRIPTION
**Pull request by adairrr**
This change makes it so that internal actions like `UpdateModuleAddresses` are explicitly executed by the user by using `Binary` instead of the struct variant. It also updates the `CallerNotFactory` error to `CallerNotModuleFactory` to avoid ambiguity with the account factory.

### CODE REVIEW

**Comment by adairrr** [comment]
Task linked: [CU-3f25exp Audit Manual module management endpoint](https://app.clickup.com/t/3f25exp)

**Comment by CyberHoward** [comment]
Converting back to draft, we need a way to update the dependency store as well

**Comment by adairrr** [comment]
And the InternalConfigAction should be non-exhaustive

**Comment by adairrr** [comment]
Converting back to a normal PR with the `InternalConfigAction` made non-exhaustive.

File: contracts/account/manager/src/commands.rs [hunk 1]
```diff
@@ -709,7 +709,7 @@ pub fn update_internal_config(deps: DepsMut, info: MessageInfo, config: Binary)
             // required to add Proxy after init by Account Factory.
             ACCOUNT_FACTORY
                 .assert_admin(deps.as_ref(), &info.sender)
-                .or_else(|_| OWNER.assert_admin(deps.as_ref(), &info.sender))?;
+                .or_else(|_| cw_ownable::assert_owner(deps.storage, &info.sender))?;
             update_module_addresses(deps, to_add, to_remove)
         }
```
Comment (CyberHoward) [inline contracts/account/manager/src/commands.rs:714]: You need to match with a catch all (because of the non-exhaustive)

**Review by CyberHoward — APPROVED** [review]

**Pull request merged**